### PR TITLE
chore(ui-react-geo): use latest versions of deps and some cleanup

### PR DESCRIPTION
**Pull request by sreeramsama**
#### Description of changes
Use latest versions of `ui`, `ui-react` and `ui-react-core` and some cleanup

#### Description of how you validated changes
- Ran tests locally
#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-geo/jest.config.js [hunk 1]
```diff
@@ -2,10 +2,10 @@ module.exports = {
   collectCoverage: true,
   collectCoverageFrom: ['<rootDir>/src/**/*.(ts|tsx)'],
   // ignore coverage for top level "export", version and style files
```
Comment (calebpollman) [inline packages/react-geo/jest.config.js:4]: Can we remove version here as well?
```suggestion
  // ignore coverage for top level "export" and style files
```

File: packages/react-geo/jest.config.js [hunk 2]
```diff
@@ -2,10 +2,10 @@ module.exports = {
...
-  coveragePathIgnorePatterns: ['<rootDir>/src/(index|styles|version).(ts|tsx)'],
+  coveragePathIgnorePatterns: ['<rootDir>/src/(index|styles).(ts)'],
```
Comment (calebpollman) [inline packages/react-geo/jest.config.js:5]: The parens around `ts` is unnecessary?

Comment (sreeramsama) [reply packages/react-geo/jest.config.js]: Did not notice that. Removed.

File: packages/react-geo/jest.config.js (hunk 1 above)
Comment (sreeramsama) [reply packages/react-geo/jest.config.js]: Did not notice that. Removed.

**Review by zchenwei — APPROVED** [review]

File: .changeset/metal-parrots-smoke.md
Comment (sreeramsama) [inline .changeset/metal-parrots-smoke.md:1]: Updated changeset to set major release for `ui-react-geo`

**Review by reesscot — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**